### ISSUE TITLE
Potential Python implementation

### ISSUE THREAD

**Issue by juanitorduz** [FMenchetti/CausalArima#22]
Hi! This is more like a suggestion on how a potential python implementation could look like. 

1. The package [`pycausalimpact`](https://github.com/WillianFuks/causalimpact/) is a nice python implementation of the CausalImpact approach based on the [`UnobservedComponets`](https://www.statsmodels.org/stable/generated/statsmodels.tsa.statespace.structural.UnobservedComponents.html) model of [`statsmodels`](https://www.statsmodels.org/). This implementation is not bully bayesian and relies on Maximum Likelihood Estimation. 

2. This package uses the [`UnobservedComponets`](https://www.statsmodels.org/stable/generated/statsmodels.tsa.statespace.structural.UnobservedComponents.html) model [here](https://github.com/WillianFuks/causalimpact/blob/master/causalimpact/main.py#L327). The idea is to replace or extend it to use the [`SARIMAX`](https://www.statsmodels.org/stable/generated/statsmodels.tsa.statespace.sarimax.SARIMAX.html#statsmodels.tsa.statespace.sarimax.SARIMAX) model, which both are part of `statsmodels.tsa.statespace`. 

3. Note that there is no "auto-arima" in [`statsmodels`](https://www.statsmodels.org/), but python users often just couple it with the [`AutoARIMA.`](http://alkaline-ml.com/pmdarima/modules/generated/pmdarima.arima.AutoARIMA.html#pmdarima-arima-autoarima)  function from the [`pmdarima`](http://alkaline-ml.com/pmdarima/) package.

I will be happy so help, but at the moment I do not have the capacity to port `CausalArima` to python. Nevertheless, maybe this suggestion triggers some initial draft implementations :)

**Comment by palmierieugenio** [comment]
Hi Juan, thank you for your suggestions!

We would like to start working on a first draft of a python implementation, as soon as we have a more stable version of our R package, hopefully at the beginning of next year.

We could use this function [statsmodels.tsa.arima.model.ARIMAResults.simulate](https://www.statsmodels.org/dev/generated/statsmodels.tsa.arima.model.ARIMAResults.simulate.html) to run the bootstrap simulations, extracting the needed parameters from a `pmdarima` or we could implement our version of bootstrap. For the closed formula solution, ` pmdarima` should be enough.

Anyway, if you or other people want to help us, we are more than happy!

**Comment by juanitorduz** [comment]
Yes, indeed by looking into [these lines](https://github.com/WillianFuks/causalimpact/blob/master/causalimpact/inferences.py#L122-L133) of the python package they are using simulation to estimate prediction intervals.

I think I won't be able to be an active contributor but I will be happy to review and test the code ;)
